cli/command/service/update.go

```diff
@@ -433,6 +434,47 @@ func getUpdatedSecrets(apiClient client.APIClient, flags *pflag.FlagSet, secrets
 		}
 		secrets = append(secrets, addSecrets...)
 	}
+
+	if flags.Changed(flagSecretReplace) {
+		replaceSecrets := []*swarm.SecretReference{}
+		secretRefQuery := []*types.SecretRequestOption{}
+
+		// Retrieve all of the secret1=secret2 replacement mappings and create a list of SecretRequestOption
+		replaceValues := flags.Lookup(flagSecretReplace).Value.(*opts.MapOpts).GetAll()
+		for _, value := range replaceValues {
+			secretRefQuery = append(secretRefQuery, &types.SecretRequestOption{Source: value})
+		}
+
+		// Retrieve a list of secret references with the SecretID filled in
+		fullSecretRefs, err := parseSecrets(apiClient, secretRefQuery)
+		if err != nil {
+			return nil, err
+		}
+
+		// Create a map with all of the full secretReferences
+		replSecretMap := make(map[string]*swarm.SecretReference)
+		for _, secretRef := range fullSecretRefs {
+			replSecretMap[secretRef.SecretName] = secretRef
+		}
+
+		// Go through all of the secrets we wish to add and see if we're being asked to replace it
+		for _, secret := range secrets {
+			for original, new := range replaceValues {
+				// We've been asked to replace this secret, replace the original name and ID for the new name and ID
+				if original == secret.SecretName {
+					secret.SecretName = new
+					secret.SecretID = replSecretMap[new].SecretID
+				}
+			}
+
+			// No matter if it got replaced or not, add the secret to the final list
+			replaceSecrets = append(replaceSecrets, secret)
```

Comment:
`replaceSecrets` doesn't seem necessary` since `secret` is being modified directly.